### PR TITLE
Creating `@fluentui/react` package

### PR DESCRIPTION
**Pull request by dzearing**
This PR introduces the `@fluentui/react` v7 package. This stages the new name for our current bits as we transition our code to Fluent UI.

This code currently exports bits from `office-ui-fabric-react` so if you're using Fabric today, you may change your imports with no breaking changes going forward.

Before:

```tsx
import { PrimaryButton } from 'office-ui-fabric-react';
import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
```

After:
```tsx
import { PrimaryButton } from '@fluentui/react';
import { PrimaryButton } from '@fluentui/react/lib/Button';
```

NOTE: this should not affect any code dependent on `office-ui-fabric-react` today. It will simply stage the ability for partners to refer to our LTS releases using the `@fluentui/react` package name.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12364)

### CODE REVIEW

File: packages/react/src/ChoiceGroupOption.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from 'office-ui-fabric-react/lib/ChoiceGroupOption';
```
Comment (ecraig12345) [inline packages/react/src/ChoiceGroupOption.ts]: Let's get rid of this file in the new version. ChoiceGroupOption is also exported from the ChoiceGroup file, and having it in two places causes issues like #11534.

File: packages/react/src/Dialog.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export * from 'office-ui-fabric-react/lib/Dialog';
+export { Dialog as default } from 'office-ui-fabric-react/lib/Dialog';
```
Comment (ecraig12345) [inline packages/react/src/Dialog.ts]: What is this doing here?

File: packages/react/src/Modal.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from 'office-ui-fabric-react/lib/Modal';
+export { Modal as default } from 'office-ui-fabric-react/lib/Modal';
```
Comment (ecraig12345) [inline packages/react/src/Modal.ts]: Same here

File: packages/react/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
+{
+  "name": "@fluentui/react",
+  "version": "7.105.1",
```
Comment (ecraig12345) [inline packages/react/package.json]: Be sure to check if this gets updated before you merge

File: packages/react/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "forceConsistentCasingInFileNames": true,
+    "strictNullChecks": true,
+    "noImplicitAny": true,
+    "moduleResolution": "node",
+    "preserveConstEnums": true,
+    "lib": ["es5", "dom"],
+    "types": ["jest", "webpack-env"],
+    "paths": {
+      "@uifabric/react/lib/*": ["./src/*"],
+      "@uifabric/react": ["./src"]
```
Comment (ecraig12345) [inline packages/react/tsconfig.json]:
```suggestion
      "@fluentui/react/lib/*": ["./src/*"],
      "@fluentui/react": ["./src"]
```

File: packages/react/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+let path = require('path');
+const resources = require('../../scripts/webpack/webpack-resources');
```
Comment (ecraig12345) [inline packages/react/webpack.config.js]:
```suggestion
const resources = require('@uifabric/build/webpack/webpack-resources');
```

File: packages/react/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+const ManifestServicePlugin = require('@uifabric/webpack-utils/lib/ManifestServicePlugin');
+
+const BUNDLE_NAME = 'fluent-ui-react';
```
Comment (ecraig12345) [inline packages/react/webpack.config.js]: Should this be `fluentui-react` to more closely match the package name?

File: packages/react/webpack.serve.config.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const path = require('path');
+const resources = require('@uifabric/build/webpack/webpack-resources');
+
+module.exports = resources.createServeConfig({
```
Comment (ecraig12345) [inline packages/react/webpack.serve.config.js]: Delete this file probably, since it won't work

File: packages/react/webpack.config.js (hunk 7 above)
Comment (dzearing) [reply packages/react/webpack.config.js]: interesting suggestion.

File: packages/react/README.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# [Fluent UI React](https://dev.microsoft.com/fabric)
+
+**The React-based front-end framework for building experiences for M365.**
```
Comment (ecraig12345) [inline packages/react/README.md]: Probably do this? M365 is a bit jargon-y. :) But this new version is much better overall!
```suggestion
**The React-based front-end framework for building experiences for Microsoft 365.**
```

File: packages/react/README.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+**Fluent UI 7** Summary, breaking changes, and more details available in [the wiki](https://github.com/OfficeDev/office-ui-fabric-react/wiki/Fabric-7).
+
+Fluent UI React is a collection of robust React-based components designed to make it simple for you to create consistent web experiences using the Office Design Language.
```
Comment (ecraig12345) [inline packages/react/README.md]:
```suggestion
Fluent UI React is a collection of robust React-based components designed to make it simple for you to create consistent web experiences using the Fluent Design Language.
```

Comment (ecraig12345) [reply packages/react/README.md]: Bonus: add a link to the fluent design site

File: packages/react/src/version.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+// @uifabric/react@0.1.0
```
Comment (ecraig12345) [inline packages/react/src/version.ts]: Guessing this file will be overwritten/corrected when the real version is generated, but wouldn't hurt to update it for initial checkin as well.

Comment (dzearing) [reply packages/react/src/version.ts]: maybe i should also change the scope name. :)

File: packages/react/package.json [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  "description": "Reusable React components for building experiences for M365.",
```
[the remark this answers is not in the document]
Comment (dzearing) [reply packages/react/package.json]: Resolved with 
  "description": "Reusable React components for building web experiences.",

File: packages/react/src/ChoiceGroupOption.ts (hunk 1 above)
Comment (dzearing) [reply packages/react/src/ChoiceGroupOption.ts]: That would be a breaking change.

File: packages/react/src/Dialog.ts (hunk 2 above)
Comment (dzearing) [reply packages/react/src/Dialog.ts]: Honestly not sure why we have a default export here.

**Pull request merged**